packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -252,6 +255,29 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
             ...this.connectionManager.connectionProps,
         };
     }
+  /**
+     * Returns set of props that can be logged in telemetry that provide some insights / statistics
+     * about current or last connection (if there is no connection at the moment)
+    */
+    public connectionProps(): ITelemetryProperties {
+        const common = {
+            sequenceNumber: this.lastSequenceNumber,
+            connectionMode: this.lastConnectionMode,
+        };
+        if (this.connection !== undefined) {
+            this.lastConnectionMode = this.connectionMode;
```

Comment:
I'd not update it here. setupNewSuccessfulConnection() is much better place to do so.
Please note that you will have conflict with changes I just pushed yesterday. Please merge - some code moved here to ConnectionManager.ts

Reply:
BTW, I'd say that recording relayServiceAgent on disconnect is interesting too.
So maybe right approach is to have this._connectionProps that is empty initially, but is filled in setupNewSuccessfulConnection() with properties like connectionMode, relayServiceAgent, maybe more in future. And then this function (or rather ConnectionManager.connectionProps() simply adds it on every call)

Reply:
Also updating it here means the value for first connection will be wrong. It will record what you captured in constructor, which is not guaranteed to be what service actually returned to us.